app/models/student.rb

```diff
@@ -1,8 +1,9 @@
 class Student < ActiveRecord::Base
+  belongs_to :title
   validates :first_name, :last_name, :email, :birth_date, :gender, presence: true
   validates :email, format: {with: URI::MailTo::EMAIL_REGEXP}, uniqueness: true
 
   def full_name
-    [title.presence, first_name, middle_name, last_name].compact.join(" ")
+    [title&.name.presence, first_name, middle_name, last_name].compact.join(" ")
```

Comment:
Consider adding a `to_s` method to the `Title` class that returns the `name`.

```
def to_s
  title # or other
end
```

And then we can just do

```
    [title.presence, first_name, middle_name, last_name].compact.join(" ")
```